setUpApplicationDefaults.js

```diff
@@ -14,11 +15,19 @@ import httpGracefulShutdown from 'http-graceful-shutdown';
 
 //TODO: better way to handle asynchrony here?
 
-export default async (app, config) => {
+export default async (app, config, existingLogger) => {
     let server;
     let metricsServer;
 
-    app.use(pino({
+    if (existingLogger) {
+      // we should probably check if this is a pino logger, not sure we support other types of loggers
+    }
+
+    const logger = existingLogger || pino();
+    app.logger = logger;
+
+    const loggerMiddleware = pinoHttp({
+        logger,
```

Comment:
Have you tested this? I have some recollection that passing a logger and passing serializers to pino-http are mutually wxclusive/will not work as expected?

Reply:
Yes - it works no problem. I added it to the example, and i can do requests with curl.

Reply:
And errors are serialized correctly?

Reply:
E.g. nested errors in `cause` fields

Reply:
Ah yes it will work, it calls logger.child: https://github.com/pinojs/pino-http/blob/41cec94a0d1bc7faf895bbf7852aaef872b3e4e9/logger.js#L194